### PR TITLE
Temporary implementation of Canvas.MeasureText

### PR DESCRIPTION
**Pull request by paulrouget**
@jdm: that will help with the babylonjs demos. Another approach is to implement that in JS and add it to the demos.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/24261)

### CODE REVIEW

**Comment by paulrouget** [comment]
The failure is very explicit :)

Gonna disable TextMetrics, not just `measureText` if the pref is off.

**Comment by paulrouget** [comment]
@jdm would you be ok with something like this?

**Comment by jdm** [comment]
Yes, bit let's hook it up to the existing pref that controls the fake fillText implementation.

**Comment by paulrouget** [comment]
@jdm r?

File: tests/wpt/mozilla/meta/MANIFEST.json [hunk 1]
```diff
@@ -7658,9 +7658,6 @@
     ]
    ]
   },
-  "stub": {
-   "css/blockify_inline_element.html": []
-  },
```
Comment (paulrouget) [inline tests/wpt/mozilla/meta/MANIFEST.json]: Why is `./mach update-manifest` updating that test?

Comment (jdm) [reply tests/wpt/mozilla/meta/MANIFEST.json]: It's fixing previously-incorrect updates.

**Comment by jdm** [comment]
<details>

```
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: existence and properties of interface object", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194160, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface object length", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194161, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface object name", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194162, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: existence and properties of interface prototype object", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194163, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: existence and properties of interface prototype object's \"constructor\" property", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194164, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: existence and properties of interface prototype object's @@unscopables property", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194165, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute width", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194166, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute actualBoundingBoxLeft", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194167, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute actualBoundingBoxRight", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194168, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute fontBoundingBoxAscent", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194169, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute fontBoundingBoxDescent", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194170, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute actualBoundingBoxAscent", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194171, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute actualBoundingBoxDescent", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194172, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute emHeightAscent", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194173, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute emHeightDescent", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194174, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute hangingBaseline", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194175, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute alphabeticBaseline", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194176, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute ideographicBaseline", 
    "test": "/html/dom/idlharness.worker.html", 
    "line": 194177, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "CanvasRenderingContext2D interface: operation measureText(DOMString)", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272472, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "CanvasRenderingContext2D interface: document.createElement(\"canvas\").getContext(\"2d\") must inherit property \"measureText(DOMString)\" with the proper type", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272576, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "CanvasRenderingContext2D interface: calling measureText(DOMString) on document.createElement(\"canvas\").getContext(\"2d\") with too few arguments must throw TypeError", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272577, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: existence and properties of interface object", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272637, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface object length", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272638, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface object name", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272639, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: existence and properties of interface prototype object", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272640, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: existence and properties of interface prototype object's \"constructor\" property", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272641, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: existence and properties of interface prototype object's @@unscopables property", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272642, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute width", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272643, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute actualBoundingBoxLeft", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272644, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute actualBoundingBoxRight", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272645, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute fontBoundingBoxAscent", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272646, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute fontBoundingBoxDescent", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272647, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute actualBoundingBoxAscent", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272648, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute actualBoundingBoxDescent", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272649, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute emHeightAscent", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272650, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute emHeightDescent", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272651, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute hangingBaseline", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272652, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute alphabeticBaseline", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272653, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "TextMetrics interface: attribute ideographicBaseline", 
    "test": "/html/dom/idlharness.https.html?exclude=(Document|Window|HTML.*)", 
    "line": 272654, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "The TextMetrics interface object should be exposed.", 
    "test": "/workers/semantics/interface-objects/001.worker.html", 
    "line": 331266, 
    "action": "test_result", 
    "expected": "FAIL"
}
```
</detail>

**Pull request merged**